04_Memory_Management/01_Overview.md

```diff
@@ -17,24 +17,43 @@ We will cover the following topics:
 
 | Memory Management Layer | Description |
 |---|------|
 | Physical Memory Manager | Responsible for keeping track of which parts of the available hardware memory (usually ram) are free/in-use. It usually allocates in fixed size blocks, the native page size. This is 4096 bytes on x86.|
-| Paging | It introduces the concepts of *virtual memory* and *virtual addresses*, providing the OS with a bigger address space, protection to the data and code in its pages, and isolation between programs. | 
+| Paging | It introduces the concepts of *virtual memory* and *virtual addresses*, providing the OS with a bigger address space, protection to the data and code in its pages, and isolation between programs. |
 | Virtual memory manager | For a lot of projects, the VMM and paging will be the same thing. However the VMM should be seen as the virtual memory *manager*, and paging is just one tool that it uses to accomplish its job: ensuring that a program has memory where it needs it, when it needs it. Often this is just mapping physical ram to the requested virtual address (via paging or segmentation) |
-| Heap Allocator | The VMM can handle page-sized allocations just fine, but that is not always useful. A heap allocator allows for allocations of any size, big or small. | 
+| Heap Allocator | The VMM can handle page-sized allocations just fine, but that is not always useful. A heap allocator allows for allocations of any size, big or small. |
+
+## A Word of Wisdom
+
+As said at the beginning of this chapter the  Memory Management Unit, is a big area in the kernel, but this is not the only characteristic. Another important one is that it is going to be used directly and/or indirectly virtually everywhere. So while designing it we should take extra care on all its implication and ramification. It mostly depends on how we design our kernel, and what features we want to have, so spending some time now thinking of what we want to achieve, can save us a lot of headaches, and rewrites/changes to mmu in the future.
 
 ## PMM - Physical Memory Manager
 
-- Exists at a system-level, there is only a single PMM per running os, and it manages all of the available memory. 
+The main features of a PMM are:
+
+- Exists at a system-level, there is only a single PMM per running os, and it manages all of the available memory.
   - There are implementation that use a co-operative pmm per cpu core design, which move excess free pages to other pmms, or request pages from other pmms if needed. These can be extremely problematic if not designed properly, and should be considered an advanced topic.
 - Keeps track of whether a page is currently in use, or free.
 - Is responsible for protecting non-usable memory regions from being used as general memory (mmio, acpi or bootloader memory).
 
+usually this is the lowest level of allocation, and only the kernel should access/use it.
```

Comment:
Capitalize `usually` for consistency :)
I think we can just keep one of `access/use` here